### ISSUE TITLE
Add support for paginated storage queries

### ISSUE THREAD

**Issue by monitz87** [paritytech/substrate-api-sidecar#562]
**Proposed Change or Idea**

Right now the `/pallets/:palletId/storage/:storageItemId` route only offers the ability to query a storage item passing all required parameters. It would be useful to be able to query the entire map (or double map) using pagination (the equivalent of polkadot.js `.entriesPaged`

**Interface**

The natural choice for me would be `/pallets/:palletId/storage/:storageItemId/entries`, with the following query params:

- arg: the first key for double maps (optional in case of a double map query, should probably throw an error if passed for a single map query)
- size: amount of entries to be queried (pageSize)
- start: key from which to start (startKey)

The response should include the queried data, and the startKey for the next page

**Comment by emostov** [comment]
cc// @TarikGul @dvdplm

**Comment by TarikGul** [comment]
I think an endpoint like this is totally reasonable, and definitely holds merit. Other than some small optimization's such as a pageSize limit I think the above structure is good.

**Comment by karen-cb** [comment]
+1 on this feature request. 

Here's some additional info regarding my use case:

Retrieving entries is very commonly-used in polkadot.js. 
For example, the code snippet below retrieves all stakers/exposures for era 1073. 
```
const eraStakers = await api.query.staking.erasStakers.entries(1073)
```

With sidecar API, users are required to specify all keys, for example: 
`<endpoint>/pallets/staking/storage/erasStakers?keys[]=<era>&keys[]=<pubkey>`

If I leave the second key blank `pallets/staking/storage/erasStakers?keys[]=1073&keys[]=`, key2 passed in as `""` (empty string), resulting in the following response:
```
{
    "at": {
        "hash": "0x18537d3d62996f858bcf27ba023ffccc05e519f98bff7594f50e4f5ae0f01596",
        "height": "15483245"
    },
    "pallet": "staking",
    "palletIndex": "7",
    "storageItem": "erasStakers",
    "keys": [
        "1073",
        ""
    ],
    "value": {
        "total": "0",
        "own": "0",
        "others": []
    }
}
```

This is because keys are passed directly into the [function arguments](https://github.com/paritytech/substrate-api-sidecar/blob/master/src/services/pallets/PalletsStorageService.ts#L71).

I think OP's original proposed change was well-thought-out. Adding the ability to query entries with pagination support would be amazing.

**Comment by TarikGul** [comment]
@karen-cb I am more than happy adding this, we recently reworked a lot of the internals for all the pallets endpoints so I think this would also be a good time to add this. I also think we could get this in relatively soon.

@marshacb @IkerAlus What do we think about adding this to the queue. I think it's pretty "trivial" overall. Let's move forward with the original idea of `/pallets/:palletId/storage/:storageItemId/entries`, and then just attach `keys` as a query param as we have normally.

**Comment by IkerAlus** [comment]
@TarikGul yes, I believe we considered this time ago. If you think it will be trivial I suggest to add it. Perhaps, @Imod7 is interested in taking this one?
